combine/usergroup_test.go

```diff
@@ -0,0 +1,66 @@
+package combine
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestByUserGroupFiles(t *testing.T) {
+	Convey("Given byusergroup files and an output", t, func() {
+		inputs, output, outputPath := buildByUserGroupFiles(t)
+
+		err := MergeUserGroupFiles(inputs, output)
+		So(err, ShouldBeNil)
+
+		Convey("You can merge and compress the byusergroup files to the output", func() {
+			_, err := os.Stat(outputPath)
+			So(err, ShouldBeNil)
+		})
+
+		Convey("The proper content exists within the output file", func() {
+			actualContent, err := ReadCompressedFile(outputPath)
+			So(err, ShouldBeNil)
+
+			expectedContent := "KMace34\tkyle\ttest/dir/\t6\t9\n"
```

Comment:
As for the other filetype test, needs to be more complete with input that has multiple files per user&group combo.